src/freenet/node/useralerts/PeersOffersUserAlert.java

```diff
@@ -0,0 +1,85 @@
+package freenet.node.useralerts;
+
+import freenet.clients.http.ConnectionsToadlet;
+import freenet.config.InvalidConfigValueException;
+import freenet.config.NodeNeedRestartException;
+import freenet.l10n.NodeL10n;
+import freenet.node.Node;
+import freenet.support.HTMLNode;
+
+public class PeersOffersUserAlert extends AbstractUserAlert {
+
+    private final String frefFiles;
+
+    private final Node node;
+
+    public PeersOffersUserAlert(String frefFiles, Node node) {
+        this.frefFiles = frefFiles;
+        this.node = node;
+    }
+
+    @Override
+    public String getTitle() {
+        return l10n("title");
+    }
+
+    @Override
+    public HTMLNode getHTMLText() {
+        HTMLNode content = new HTMLNode("div");
+        content.addChild("p", l10n("text"));
+        content.addChild("p", frefFiles);
+        HTMLNode form = content.addChild("form",
+                new String[] {"action", "method"},
+                new String[] {"/friends/", "post"});
+        form.addChild("input",
+                new String[] {"type", "name", "value"},
+                new String[] {"hidden", "formPassword", node.clientCore.formPassword});
+        form.addChild("input",
+                new String[] {"type", "name", "value"},
+                new String[] {"hidden", "peers-offers-files", "true"});
+
+        ConnectionsToadlet.drawPeerTrustInputForAddPeerBox(form);
```

Comment:
Looks like this method should not live in that particular toadlet if it’s used from here as well.

Reply:
I processed all the comments except for the issue of moving the `nodeConfig` functionality from the `Node` class. Thank you for correcting me!